src/RequestSpec.ts

```diff
@@ -0,0 +1,34 @@
+interface NegatableOption {
+  value: string;
+  negated: boolean;
+}
+
+export interface RequestSpec {
+  host_name: string;
+  host_name_regex: NegatableOption;
+  site: string;
+  service: string;
+  service_regex: NegatableOption;
+  host_in_group: NegatableOption;
+  host_labels: string[];
+  service_in_group: NegatableOption;
+  host_tags: Array<{ group: string; tag: string; operator: 'is' | 'isnot' }>;
+  aggregation: 'lines' | 'sum' | 'average' | 'min' | 'max';
+  graph_type: 'template' | 'metric';
+  graph: string;
+}
+
+export const defaultRequestSpec: RequestSpec = {
+  host_name: '',
```

Comment:
can you check this out with "host_name" to see if it would be much work? if it's too much work, remove the change and we forget about it, otherwise we keep it and add a TODO to do the same with the other properties?